### PR TITLE
Add Vector2/3/4 tests

### PR DESCRIPTION
**Pull request by dr-vortex**
Also adds tmp and temp to .gitignore

Note that this is to make debugging #13699 easier.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13973/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13973/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13973/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13973/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13973/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13973/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13973/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13973/merge#BCU1XR#0

**Review by Popov72 — APPROVED** [review]
Thanks a lot for this!

File: packages/dev/core/test/unit/Math/babylon.math.vector.test.ts [hunk 1]
```diff
@@ -1,61 +1,1240 @@
...
+                expect(vector1.y).toBe(6);
+            });
+        });
+
+        describe("divide", () => {
+            it("should divide the vector by another vector", () => {
+                const result = vector1.divide(vector2);
+                expect(result.x).toBeCloseTo(1 / 3);
+                expect(result.y).toBeCloseTo(2 / 4);
+            });
```
Comment (carolhmj) [inline packages/dev/core/test/unit/Math/babylon.math.vector.test.ts:86]: Same, checking if vector1 and vector2 are not changed would be nice

**Review by carolhmj — COMMENTED** [review]
Very nice! In general, I think it's possible to make the tests cover even more if you check that the vectors used for operations haven't changed in the appropriate case. You could create a boolean variable, set it to true/false in the case it should have changed/not, and check on an after() function according to the variable.

File: packages/dev/core/test/unit/Math/babylon.math.vector.test.ts [hunk 2]
```diff
@@ -1,61 +1,1240 @@
...
+                expect(vector1.y).toBe(6);
+            });
+        });
+
+        describe("addToRef", () => {
+            it("should add the given vector to the current vector and store the result in the given output vector", () => {
+                const result = new Vector2();
+                vector1.addToRef(vector2, result);
+                expect(result.x).toBe(4);
+                expect(result.y).toBe(6);
```
[the remark this answers is not in the document]
Comment (dr-vortex) [reply packages/dev/core/test/unit/Math/babylon.math.vector.test.ts:47]: Added in [75a2f6c](https://github.com/BabylonJS/Babylon.js/pull/13973/commits/75a2f6c7681087089282cc29d62f97520f590ea8)

File: packages/dev/core/test/unit/Math/babylon.math.vector.test.ts [hunk 3]
```diff
@@ -1,61 +1,1240 @@
...
+        beforeEach(() => {
+            vector1 = new Vector2(1, 2);
+            vector2 = new Vector2(3, 4);
+        });
+
+        describe("add", () => {
+            it("should add two vectors and return a new vector", () => {
+                const result = vector1.add(vector2);
+                expect(result.x).toBe(4);
+                expect(result.y).toBe(6);
```
[the remark this answers is not in the document]
Comment (dr-vortex) [reply packages/dev/core/test/unit/Math/babylon.math.vector.test.ts:29]: Added in [75a2f6c](https://github.com/BabylonJS/Babylon.js/pull/13973/commits/75a2f6c7681087089282cc29d62f97520f590ea8)

File: packages/dev/core/test/unit/Math/babylon.math.vector.test.ts (hunk 1 above)
Comment (dr-vortex) [reply packages/dev/core/test/unit/Math/babylon.math.vector.test.ts:103]: Added in [75a2f6c](https://github.com/BabylonJS/Babylon.js/pull/13973/commits/75a2f6c7681087089282cc29d62f97520f590ea8)

**Review by carolhmj — APPROVED** [review]
Love love it!

**Pull request merged**